framework/tlvf/src/src/CmduMessage.cpp

```diff
@@ -15,7 +15,7 @@ eMessageType CmduMessage::getMessageType()
     uint16_t msgValue = 0;
     auto cmduhdr      = getCmduHeader();
     msgValue          = (uint16_t)cmduhdr->message_type();
-    if (cmduhdr->is_finalized())
+    if (cmduhdr->is_finalized() || is_swapped())
```

Comment:
commit message says "prevent double swapping" but the condition here is met if it is already swapped.

couldn't suggest an alternative as I am not sure what the actual condition should be:
`(x && y)`
or
`(x && !y)`
or
`(x || !y)` 

same to the rest of occurences

Reply:
Maybe the commit message is misleading.
These are getter functions, and they actually return a "swapped back" copy of the value in case the message was finalized or swapped.